.hooks/post-checkout

```diff
@@ -12,3 +12,5 @@ if [ -f pnpm-lock.yaml ]; then
 		pnpm_helper
 	fi
 fi
+
+submodule_helper
```

Comment:
The current default behavior works well for basic submodule usage, like the pandora folder is even hidden from vscode inside the assets repo, and does everything auto-magically.

I have already added a similar setup to assets repo with `SKIP_SUBMODULE_UPDATE` option, although that change was pure as it didn't involve any change other than offering extra options.

This change is the other side of that change as this allows a simple `git checkout` inside pandora submodule folder to do everything at once

For me, this flow is much nicer than the forced way of doing things:
no git action overrides the content of the submodule
checking out a submodule is the only action you need to do if you want it to be in a specific state
`git submodule update` works as expected

I will add some comments for this, as it is meant for advanced git usage and for those who like this more in a manual way

Currently, I do not really see any need to add an option to provide hook overrides as auto-magic, and the now new manual way is well defined and no further hooks should be required for them

Reply:
Comment only, or lift it out to separate PR?

Reply:
So this is intended for usage in asset repo? The usage I intended for asset repo was not to use submodule there, but for submodule to simply be a way to install common without having to do so manually. If one plans to develop concurrently for asset repo and main repo, I would strongly recommend using `pnpm link` instead of working inside the submodule.
Also the hiding of the submodule is thanks to settings I put into VSCode for asset repo with this usecase in mind.

I would prefer for this to be lifted to separate PR where it can be discussed further, possibly also about how to simplify this in general.